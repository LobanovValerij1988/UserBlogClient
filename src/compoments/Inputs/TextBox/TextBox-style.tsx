import {createStyles, Theme} from "@material-ui/core";

export const style = (theme:Theme) => createStyles({
   textBox: {
       width: "100%"
   }
});
