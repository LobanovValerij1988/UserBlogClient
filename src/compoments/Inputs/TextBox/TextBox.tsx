import React  from "react";
import {withStyles, WithStyles} from "@material-ui/styles";
import {style} from "./TextBox-style";
import {TextField} from "@material-ui/core";

interface ITextBox extends WithStyles<typeof style> {
HasError: boolean,
ErrorMessage: string,
label: string,
placeholder: string,
value: string,
setValue: (value: string) => void,
type:string,
multiline?: boolean
rows?: number
}

const TextBox = withStyles(style)(
    ({classes,
                HasError,
                ErrorMessage,
                label,
                placeholder,
                value,
                setValue,
                type,
                multiline = false,
                rows = 1,
              }: ITextBox) => {
  return(
        <TextField error = {HasError}
                   multiline={multiline}
                   rows={rows}
                   type = {type}
                   helperText = {HasError ? ErrorMessage : ""}
                   label = { label}
                   placeholder = {placeholder}
                   margin = {"normal"}
                   className = {classes.textBox}
                   value = {value}
                   onChange = {(e) => {setValue(e.target.value)}}
        />
    )
})

export default TextBox;