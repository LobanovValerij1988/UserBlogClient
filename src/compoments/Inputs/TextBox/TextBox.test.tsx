import { render, screen } from '@testing-library/react';
import TextBox from "./TextBox";
import userEvent from "@testing-library/user-event";

const errorMessage = "something was wrong"
const typedMesage = "react"
const onChange = jest.fn()

describe('TextBox component', () =>{
     it('Error message  show', () => {
         render(<TextBox HasError={true} label = "our label" ErrorMessage = {errorMessage}  placeholder = ""  value = ""  setValue = {onChange} type = "text"  />) ;
         expect(screen.getByText(errorMessage)).toBeInTheDocument();
     })
     it('Error message doesnt show', () => {
            render(<TextBox HasError={false} label = "our label" ErrorMessage = {errorMessage}  placeholder = ""  value = ""  setValue = {onChange} type = "text"  />) ;
            expect(screen.queryByText(errorMessage)).toBeNull();
     })
    it('On change works correctly', () => {
             render(<TextBox HasError={false} label = "our label" ErrorMessage = {errorMessage}  placeholder = ""  value = ""  setValue = {onChange} type = "text"  />) ;
             userEvent.type(screen.getByRole('textbox'), typedMesage)
             expect(onChange).toHaveBeenCalledTimes(typedMesage.length)

    })
   it("Text Box snapshot", ()=>{
       const textBox =  render(<TextBox HasError={false} label = "our label" ErrorMessage = {errorMessage}  placeholder = ""  value = ""  setValue = {onChange} type = "text"  />) ;
       expect(textBox).toMatchSnapshot();
   })
    })