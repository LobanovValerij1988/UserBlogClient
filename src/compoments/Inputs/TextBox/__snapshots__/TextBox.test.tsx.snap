// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox component Text Box snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root Component-textBox-4 css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
        >
          our label
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            placeholder=""
            rows="1"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-2s9zix"
            >
              <span>
                our label
              </span>
            </legend>
          </fieldset>
        </div>
        
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root Component-textBox-4 css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="false"
      >
        our label
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          placeholder=""
          rows="1"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-2s9zix"
          >
            <span>
              our label
            </span>
          </legend>
        </fieldset>
      </div>
      
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
